Jenkinsfile

```diff
@@ -1,154 +1,64 @@
-        build: env.BUILD_TAG,
-        public: 'public restricted'
-    ]
-    def capabilities = defaultCapabilities.clone()
-    capabilities.putAll(desiredCapabilities)
-    def json = JsonOutput.toJson([capabilities: capabilities])
-    writeFile file: 'capabilities.json', text: json
-}
-
-/** Run Tox
- *
- * @param environment test environment to run
-*/
-def runTox(environment) {
-  def options = env.PYTEST_ADDOPTS ?: ''
-  def processes = env.PYTEST_PROCESSES ?: 'auto'
-  try {
-    wrap([$class: 'AnsiColorBuildWrapper']) {
-      withCredentials([[
-        $class: 'StringBinding',
-        credentialsId: 'SAUCELABS_API_KEY',
-        variable: 'SAUCELABS_API_KEY']]) {
-        withEnv(["PYTEST_ADDOPTS=${options} " +
-          "-n=${processes} " +
-          "--driver=SauceLabs " +
-          "--variables=capabilities.json " +
-          "--color=yes"]) {
-          sh "tox -e ${environment}"
-        }
-      }
-    }
-  } catch(err) {
-    currentBuild.result = 'FAILURE'
-    throw err
-  } finally {
-    dir('results') {
-      stash environment
-    }
+pipeline {
+  agent any
+  options {
+    ansiColor('xterm')
+    timestamps()
+    timeout(time: 1, unit: 'HOURS')
   }
-}
-
-/** Send a notice to #fxtest-alerts on irc.mozilla.org with the build result
- *
- * @param result outcome of build
-*/
-def ircNotification(result) {
-  def nick = "fxtest${BUILD_NUMBER}"
-  def channel = '#fx-test-alerts'
-  result = result.toUpperCase()
-  def message = "Project ${JOB_NAME} build #${BUILD_NUMBER}: ${result}: ${BUILD_URL}"
-  node {
-    sh """
-        (
-        echo NICK ${nick}
-        echo USER ${nick} 8 * : ${nick}
-        sleep 5
-        echo "JOIN ${channel}"
-        echo "NOTICE ${channel} :${message}"
-        echo QUIT
-        ) | openssl s_client -connect irc.mozilla.org:6697
-    """
+  environment {
+    SAUCELABS_API_KEY = credentials('SAUCELABS_API_KEY')
   }
-}
-
-stage('Checkout') {
-  node {
-    timestamps {
-      deleteDir()
-      checkout scm
-      stash 'workspace'
-    }
-  }
-}
-
-stage('Lint') {
-  node {
-    timestamps {
-      deleteDir()
-      unstash 'workspace'
-      sh 'tox -e flake8'
+  stages {
+    stage('Lint') {
+      steps {
+        sh "tox -e flake8"
+      }
     }
-  }
-}
-
-try {
-  stage('Test') {
-    node {
-      timeout(time: 1, unit: 'HOURS') {
-        timestamps {
-          deleteDir()
-          unstash 'workspace'
-          try {
-            writeCapabilities(capabilities)
-            runTox(environment)
-          } catch(err) {
-            currentBuild.result = 'FAILURE'
-            throw err
-          } finally {
-            dir('results') {
-              stash environment
-            }
-          }
+    stage('Test') {
+      environment {
+        /** See https://issues.jenkins-ci.org/browse/JENKINS-42857 - we'd like to expand this out into multi-line concatenations */
+        PYTEST_ADDOPTS = "--color=yes --driver=SauceLabs --variables=capabilities.json"
```

Comment:
Put this in the main `environment` section on line 19. We only need to have two sections when we're using a secret variables file.